app/src/main/java/nerd/tuxmobil/fahrplan/congress/repositories/AppRepository.kt

```diff
@@ -304,6 +309,12 @@ object AppRepository {
         alarmsDatabaseRepository.update(values, alarm.eventId)
     }
 
+    fun readHighlightEventIds() = readHighlights()
+            .asSequence()
+            .filter { it.isHighlight }
+            .map { it.eventId.toString() }
+            .toSet()
```

Comment:
The number of intermediate lists this would create combined with the fact that due to the filter at least one collection can't be allocated with the correct size in both cases anyway.

Without `asSequence()`:
- `filter()` starts with an `ArrayList()` that might have to be resized multiple times depending on how many items are put into it
- `map()` creates a new list with the correct size
- `toSet()` creates a new set with the correct size

Allocations: 3 collections + list resizing

With `asSequence()`:
- `asSequence()` creates a `Sequence` instance
- `filter()` creates a `FilterSequence` instance
- `map()` creates a `TransformingSequence` instance
- `toSet()` creates a new set that might have to be resized multiple times

Allocations: 3 sequence objects + 1 collection + set resizing

There are more allocations, but they should be mostly the same for both methods.

While it might look worse with `asSequence()`, the sequence objects are quite lightweight compared to the collections.

Reply:
Thank you. I am fine with the sequence. I guess since `readAlarmEventIds()` does one allocation less you did not consider adding it there. Please feel free to resolve this.

Reply:
For `readAlarmEventIds()` the math is different.

Without `asSequence()`:
- `map()` creates a new list with the correct size
- `toSet()` creates a new set with the correct size

Allocations: 2 collections

With `asSequence()`:
- `asSequence()` creates a Sequence instance
- `map()` creates a TransformingSequence instance
- `toSet()` creates a new set that might have to be resized multiple times

Allocations: 2 sequence objects + 1 collection + set resizing